Lib/test/test_memoryio.py

```diff
@@ -517,6 +517,28 @@ def test_relative_seek(self):
         memio.seek(1, 1)
         self.assertEqual(memio.read(), buf[1:])
 
+    def test_peek(self):
+        buf = self.buftype("1234567890")
+        with self.ioclass(buf) as memio:
+            self.assertEqual(memio.tell(), 0)
+            self.assertEqual(memio.peek(1), buf[:1])
+            self.assertEqual(memio.peek(1), buf[:1])
+            self.assertEqual(memio.peek(), buf[:1])
+            self.assertEqual(memio.peek(0), buf)
```

Comment:
Lacks tests for negative and "larger than len(buf)" sizes.

Reply:
Added